app/services/authenticator.ts

```diff
@@ -72,20 +72,30 @@ export default class AuthenticatorService extends Service {
     await this.syncCurrentUser();
   }
 
-  initiateLogin(redirectPath: string | null) {
-    const isFullUrl = redirectPath?.startsWith(`${config.x.backendUrl}`);
-
-    if (isFullUrl) {
-      window.location.href = `${config.x.backendUrl}/login?next=` + redirectPath;
-    } else if (redirectPath) {
-      window.location.href = `${config.x.backendUrl}/login?next=` + encodeURIComponent(`${window.origin}${redirectPath}`);
-    } else if (this.router.currentURL) {
+  initiateLogin(redirectPathOrUrl: string | null = null) {
+    if (redirectPathOrUrl) {
+      this.initiateLoginAndRedirect(redirectPathOrUrl);
+
+      return;
+    }
+
+    if (this.router.currentURL) {
       window.location.href = `${config.x.backendUrl}/login?next=` + encodeURIComponent(`${window.origin}${this.router.currentURL}`);
     } else {
       window.location.href = `${config.x.backendUrl}/login?next=` + encodeURIComponent(window.origin);
     }
   }
 
+  initiateLoginAndRedirect(redirectPathOrUrl: string) {
+    if (redirectPathOrUrl.startsWith(`${config.x.backendUrl}`)) {
+      window.location.href = `${config.x.backendUrl}/login?next=` + redirectPathOrUrl;
+    } else if (redirectPathOrUrl.startsWith('/')) {
+      window.location.href = `${config.x.backendUrl}/login?next=` + encodeURIComponent(`${window.origin}${redirectPathOrUrl}`);
```

Comment:
Still too much branching - @ryan-gang can you take a fresh look at this and try to simplify? I'm positive this whole thing can be made far simpler than it currently is. 

Primary thing that stands out here is: Why should backend URLs be treated any differently? We only need to differentiate between "path" and "URL"